components/BookmarkButton/BookmarkButton.js

```diff
@@ -0,0 +1,32 @@
+import styled from "styled-components";
+import BookmarkIcon from "../../public/icons/bookmark";
+
+export default function BookmarkButton({
+  id,
+  onToggleBookmark,
+  restaurant,
+  userInfos,
+}) {
+  const matchedInfo = userInfos?.find((info) => info.id === id);
```

Comment:
Your button would be much more flexible and decoupled if you lift the logic up.
I would suggest to have only `isBookmarked` and `onToggleBookmark` as props and do the rest at the place where the component is used.
Why is that better?
- It decouples the logic from the view and makes the component more flexible
- It's easier to test the component because there are less dependencies
- When you map through your list to display the restaurants, it's easy to check if the current restaurant is bookmarked or not. With your solution you have to call a `find()` for every button you render.